tests/components/unifiprotect/test_sensor.py

```diff
@@ -0,0 +1,283 @@
+        unique_id, entity_id = ids_from_device_description(
+            Platform.SENSOR, sensor, description
+        )
+
+        entity = entity_registry.async_get(entity_id)
+        assert entity
+        assert entity.unique_id == unique_id
+
+        state = hass.states.get(entity_id)
+        assert state
+        assert state.state == expected_values[index]
+        assert state.attributes[ATTR_ATTRIBUTION] == DEFAULT_ATTRIBUTION
+
+    # BLE signal
+    unique_id, entity_id = ids_from_device_description(
+        Platform.SENSOR, sensor, ALL_DEVICES_SENSORS[1]
+    )
+
+    entity = entity_registry.async_get(entity_id)
+    assert entity
+    assert entity.disabled is True
+    assert entity.unique_id == unique_id
+
+    await enable_entity(hass, mock_entry.entry.entry_id, entity_id)
+
+    state = hass.states.get(entity_id)
+    assert state
+    assert state.state == "-50"
+    assert state.attributes[ATTR_ATTRIBUTION] == DEFAULT_ATTRIBUTION
+
+
+async def test_sensor_setup_nvr(
+    hass: HomeAssistant, mock_entry: MockEntityFixture, now: datetime
+):
+    """Test sensor entity setup for NVR device."""
+
+    mock_entry.api.bootstrap.reset_objects()
+    nvr: NVR = mock_entry.api.bootstrap.nvr
+    nvr.up_since = now
+    nvr.system_info.cpu.average_load = 50.0
+    nvr.system_info.cpu.temperature = 50.0
+    nvr.storage_stats.utilization = 50.0
+    nvr.system_info.memory.available = 50.0
+    nvr.system_info.memory.total = 100.0
+    nvr.storage_stats.storage_distribution.timelapse_recordings.percentage = 50.0
+    nvr.storage_stats.storage_distribution.continuous_recordings.percentage = 50.0
+    nvr.storage_stats.storage_distribution.detections_recordings.percentage = 50.0
+    nvr.storage_stats.storage_distribution.hd_usage.percentage = 50.0
+    nvr.storage_stats.storage_distribution.uhd_usage.percentage = 50.0
+    nvr.storage_stats.storage_distribution.free.percentage = 50.0
+    nvr.storage_stats.capacity = 50.0
+
+    await hass.config_entries.async_setup(mock_entry.entry.entry_id)
+    await hass.async_block_till_done()
+
+    # 2 from all, 4 from sense, 12 NVR
+    assert_entity_counts(hass, Platform.SENSOR, 12, 9)
+
+    entity_registry = er.async_get(hass)
+
+    expected_values = (
+        now.replace(second=0, microsecond=0).isoformat(),
+        "50.0",
+        "50.0",
+        "50.0",
+        "50.0",
+        "50.0",
+        "50.0",
+        "50.0",
+        "50.0",
+        "50.0",
+        "50.0",
+        "50",
+    )
+    for index, description in enumerate(NVR_SENSORS):
+        unique_id, entity_id = ids_from_device_description(
+            Platform.SENSOR, nvr, description
+        )
+
+        entity = entity_registry.async_get(entity_id)
+        assert entity
+        assert entity.disabled is not description.entity_registry_enabled_default
+        assert entity.unique_id == unique_id
+
+        if not description.entity_registry_enabled_default:
```

Comment:
Please create a separate constant for all the disabled sensors so we don't need any conditional logic in the middle of the test.